administrator/components/com_admin/script.php

```diff
@@ -2251,6 +2252,79 @@ public function postflight($action, $installer)
             return false;
         }
 
+        if (!$this->migratePrivacyconsentConfiguration()) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Migrate privacyconsents system plugin configuration
+     *
+     * @return  boolean  True on success
+     *
+     * @since   5.0.0
+     */
+    private function migratePrivacyconsentConfiguration(): bool
+    {
+        $db = Factory::getDbo();
+
+        try {
+            // Get the PrivacyConsent system plugin's parameters
+            $row = $db->setQuery(
+                $db->getQuery(true)
+                    ->select($db->quotename('enabled'), $db->quoteName('params'))
+                    ->from($db->quoteName('#__extensions'))
+                    ->where($db->quoteName('type') . ' = ' . $db->quote('plugin'))
+                    ->where($db->quoteName('folder') . ' = ' . $db->quote('system'))
+                    ->where($db->quoteName('element') . ' = ' . $db->quote('privacyconsent'))
+            )->loadObject();
+        } catch (Exception $e) {
+            echo Text::sprintf('JLIB_DATABASE_ERROR_FUNCTION_FAILED', $e->getCode(), $e->getMessage()) . '<br>';
+
+            return false;
+        }
+        // If disbled there is nothing to migrate
+        if (!$row->enabled) {
+            return true;
+        }
+
+        $params = new Registry($row->params);
+
+        // If consent expiration was disbled there is nothing to migrate
+        if (!$params->get('enabled', 0)) {
+            return true;
+        }
+
+        /** @var SchedulerComponent $component */
+        $component = Factory::getApplication()->bootComponent('com_scheduler');
+
+        /** @var TaskModel $model */
+        $model = $component->getMVCFactory()->createModel('Task', 'Administrator', ['ignore_request' => true]);
+        $task  = [
+            'title'           => 'PrivacyConsent',
+            'type'            => 'privacy.consent',
+            'execution_rules' => [
+                'rule-type'      => 'interval-hours',
+                'interval-hours' => $params->get('cachetimeout', 6),
+                'exec-time'      => gmdate("H:i", $params->get('lastrun', 0)),
+            ],
```

Comment:
```suggestion
            'execution_rules' => [
                'rule-type'     => 'interval-days',
                'interval-days' =>  $params->get('cachetimeout', 30),
                'exec-time'     => gmdate("H:i", $params->get('lastrun', 0)),
                'exec-day'      => gmdate('d'),
            ],
```
Cachetimeout was given in days in the system plugin, and default was 30.